### DISCUSSION TITLE
[OP Stack Deployment] L2 with custom gas token but getting "Wrapped Ether"

### DISCUSSION THREAD

**Discussion by ddoliveiraa** [ethereum-optimism/developers#662] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I'm working on creating an L2 using the OP Stack. 

I followed the "getting-started" guide and successfully got it running. Now, I'm trying to implement a custom gas token following the instructions in the documentation. However, when I attempt to validate the implementation with a cast, it still shows "Wrapped Ether" instead of my custom token.

Here's what I've done so far:

- Added useCustomGasToken and customGasTokenAddress to the getting-started.json.
- Carefully followed the documentation multiple times.

Despite this, I haven't been able to get it to work.

**Comment by Chomtana** [comment]
Where does it show "Wrapped Ether"?

If it's in explorer or wallet then it's the client config or the env variable. You would probably look into frontend code or the env in the explorer.

I think there is no standard way to config for the native gas token name. As a result, each framework come up with its own way.

**Reply by ddoliveiraa** [reply]
When i do "cast call --rpc-url $L2_ETH_RPC_URL 0x4200000000000000000000000000000000000006 'name()(string)'" it prints me "Wrapped Ether". And when I do the "cast call --rpc-url $L2_ETH_RPC_URL 0x4200000000000000000000000000000000000015 'isCustomGasToken()(bool)'" it prints false.

**Reply by surhud004** [reply]
Hey @ddoliveiraa I am running the same configuration setup. It returns `Wrapped Ether` for me as well. However, if you connect your chain with Metamask, you should be able to see your custom gas token integrated instead of ETH. Another way to verify this is to query your `SystemConfigProxy` contract on your L1 -

<img width="200" alt="image" src="https://github.com/user-attachments/assets/f83f2d7c-6b54-4009-b26a-d65d28a99234">

**Reply by KenianShi** [reply]
Yes, I am the same, but does this prove that my L2 deployment was successful ?

> Hey @ddoliveiraa I am running the same configuration setup. It returns `Wrapped Ether` for me as well. However, if you connect your chain with Metamask, you should be able to see your custom gas token integrated instead of ETH. Another way to verify this is to query your `SystemConfigProxy` contract on your L1 -
> 
> <img alt="image" width="200" src="https://private-user-images.githubusercontent.com/42063989/390835500-f83f2d7c-6b54-4009-b26a-d65d28a99234.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI5NjI1NDksIm5iZiI6MTczMjk2MjI0OSwicGF0aCI6Ii80MjA2Mzk4OS8zOTA4MzU1MDAtZjgzZjJkN2MtNmI1NC00MDA5LWIyNmEtZDY1ZDI4YTk5MjM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTMwVDEwMjQwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxYWNjYWU2ZTQ3ODlmNmM2YTQ4ODRmYzJjYjkyMzIxMTI2YWQ4MjQyZmU1ZTM1ODNiZWJlZjNlMzZlZTQwYmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TZ2FoivRF3CbuYQoE_cEXy2ogfGswef0Q5Y2FqHMeO0">

**Reply by KenianShi** [reply]
> When i do "cast call --rpc-url $L2_ETH_RPC_URL 0x4200000000000000000000000000000000000006 'name()(string)'" it prints me "Wrapped Ether". And when I do the "cast call --rpc-url $L2_ETH_RPC_URL 0x4200000000000000000000000000000000000015 'isCustomGasToken()(bool)'" it prints false.

Have you solved this problem? The same thing happened to me

**Reply by priyanshuthapliyal** [reply]
Hi @KenianShi @ddoliveiraa  actually same issue i am facing so i have raised this issue with optimism team as well. You can see that issue. According to my observation it is due to how we create genesis and rollup.json using op-node newer version. Because this issue is not there in v1.7.7. Anyways chain is working fine even with new version it just the cast command output we are getting is wrong for predeploys.
[https://github.com/ethereum-optimism/optimism/issues/13007](url)

**Reply by opfocus** [reply]
"This calls the L1Block predeploy should return true" check, my return was false

In addition to trying to use a newer version, it may be necessary to wait for a while for L2 synchronization to complete. The configuration on L2 may be triggered by the deployment of the L1 contract, and may need to be synchronized before you can see the expected results.

This query may be related to the synchronization of the chain, because I have stopped my chain and cannot test again for the time being. If you encounter this too, please wait for synchronization for a period of time before trying to query again.

**Comment by Satyam78612** [comment]
I want to know what you mean by being part of the superchain enables seamless transactions, we will have full control over our chain, scalability, TPS, transaction costs?

**Reply by opfocus** [reply]
please open a new discussion for your question

**Comment by Casholey** [comment]
0x1f9090aaE28b8a3dCeaDf281B0F12828e676c326

**Comment by opfocus** [comment, marked as answer]
[Deprecation of Custom Gas Tokens](https://docs.optimism.io/notices/custom-gas-tokens-deprecation#deprecation-of-custom-gas-tokens)
